### PR TITLE
WiX BA allows multiple clicks of Install button while update check is running

### PR DESCRIPTION
**Pull request by jchoover**
http://wixtoolset.org/issues/4512/

### CODE REVIEW

**Comment by heaths** [comment]
Looks okay, but wouldn't just disabling the Install button suffice until DetectComplete? That's basically what we do in our BA.

**Comment by jchoover** [comment]
Heath,
  Can't do that cleanly cause we are dog fooding the engine feed updating.  In 3.x I couldn't modify signatures, so we have to call detect a second time if the update feed fails.  I suppose we could make it conditional based off of if Detect succeeded, but it seemed more logical to break this into two different states.

File: src/Setup/WixBA/RootViewModel.cs [hunk 1]
```diff
@@ -110,23 +111,46 @@ public bool Canceled
...
             get
             {
-                return this.state;
+                return this.detectstate;
             }
 
             set
             {
-                if (this.state != value)
+                if (this.detectstate != value)
```
Comment (robmen) [inline src/Setup/WixBA/RootViewModel.cs]: Should be `detectState`? Two words?

File: src/Setup/WixBA/RootViewModel.cs [hunk 2]
```diff
@@ -110,23 +111,46 @@ public bool Canceled
...
+        }
+
+        /// <summary>
+        /// Gets and sets the installation state of the view's model.
+        /// </summary>
+        public InstallationState InstallState
+        {
+            get
+            {
+                return this.installstate;
```
Comment (robmen) [inline src/Setup/WixBA/RootViewModel.cs]: Ditto but `installState`.

Comment (jchoover) [inline src/Setup/WixBA/RootViewModel.cs]: Fixed in updated PR

File: src/Setup/WixBA/RootViewModel.cs (hunk 1 above)
Comment (jchoover) [inline src/Setup/WixBA/RootViewModel.cs]: Fixed in updated PR

File: History.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
+* JacobHoover: WIXBUG:4512 - fix WiX BA, prevent multiple install clicks
+
```
Comment (barnson) [inline History.md]: Remember the new history scheme...

**Pull request merged**